File: client/src/app/implementations/shared/models/grid.ts
import { IModel } from "src/app/interfaces/i-model";
import { generateUUID } from "three/src/math/MathUtils";
import { Vector } from "./vector";

export class Grid implements IModel 
{
    public key: string = generateUUID();
    public origin!: Vector;
    public baseX!: Vector;
    public baseY!: Vector;
    public baseZ!: Vector;
}

File: GeometryGenerationApi/Common/GeometryGeneration.Model/Model/Grid.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GeometryGeneration.Model.Model
{
    public class Grid
    {
        public string Key { get; set; }
        public Vector BaseX { get; set; }
        public Vector BaseY { get; set; }
        public Vector BaseZ { get; set; }
        public Vector Origin { get; set; }

        public Grid()
        {
            Key = Guid.NewGuid().ToString();
            BaseX = new Vector(1, 0, 0);
            BaseY = new Vector(0, 1, 0);
            BaseZ = new Vector(0, 0, 1);
            Origin = new Vector(0, 0, 0);
        }
    }
}
